### PR TITLE
Moving init?(ContentsOfFile:) to NSDictionary

### PR DESCRIPTION
**Pull request by ddunn2**
The initialiser `init?(ContentsOfFile: String)` and `init?(contentsOf: URL)` were incorrectly defined in NSMutableDictionary. According to the [docs](https://developer.apple.com/documentation/foundation/nsdictionary) it should be defined in NSDictionary instead, this PR resolves this. 

I've also updated the tests to check for this initialiser in NSDictionary rather than NSMutableDictionary.

### CODE REVIEW

**Comment by pushkarnk** [comment]
@swift-ci please test

File: Foundation/NSDictionary.swift [hunk 1]
```diff
@@ -39,6 +39,22 @@ open class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
         
         return NSGeneratorEnumerator(_storage.keys.map { _SwiftValue.fetch(nonOptional: $0) }.makeIterator())
     }
+    @available(*, deprecated)
+    public convenience init?(contentsOfFile path: String) {
+        self.init(contentsOfURL: URL(fileURLWithPath: path))
+    }
+    
+    @available(*, deprecated)
+    public convenience init?(contentsOfURL url: URL) {
```
Comment (ianpartridge) [inline Foundation/NSDictionary.swift]: The correct parameter name is `contentsOf` - see https://developer.apple.com/documentation/foundation/nsdictionary/1416069-init

File: Foundation/NSDictionary.swift [hunk 2]
```diff
@@ -39,6 +39,22 @@ open class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
         
         return NSGeneratorEnumerator(_storage.keys.map { _SwiftValue.fetch(nonOptional: $0) }.makeIterator())
     }
+    @available(*, deprecated)
```
Comment (ianpartridge) [inline Foundation/NSDictionary.swift]: Can we have a newline before this, please.

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by alblue** [comment]
@ddunn2 can these be merged now?

**Comment by ddunn2** [comment]
I've addressed the requested the changes made by @ianpartridge so assuming there's nothing else this can be merged.

**Pull request merged**